### DISCUSSION TITLE
Multiple offset values

### DISCUSSION THREAD

**Discussion by apark2459** [bitcraze/discussions#1514] [Q&A] [answered]
When I place the lighthouse deck in certain orientations, I get multiple offset values which prevent me from being able to calculate sweep angles. This happens even when I have only a single basestation turned on so I don't think it's an interference issue. Is this supposed to happen or is there something wrong with my lighthouse deck that's causing this? (note: I am interfacing the lighthouse system with a raspberry pi)

<img width="805" alt="Screenshot 2024-08-09 at 11 04 19 PM" src="https://github.com/user-attachments/assets/4768274e-6668-4738-866d-b630ee817a37">

**Comment by knmcguire** [comment]
It seems to also not be able to detect the channel sometimes. Do you have any big parts of reflections in the same room as the basestation/lighthouse deck?

**Reply by apark2459** [reply]
Thanks for the reply! There is a table and a whiteboard in tracking area that are slightly reflective. There is also glass in the room but the base stations aren't facing the glass. I personally don't think it's the reflections causing this as the signal width of the pulses are consistent (I'm guessing that reflections would have smaller pulses)

The multiple offset thing seems to happen when I angle the lighthouse deck to very specific angles. Is it possible that it's due to the fact that there are two UART interfaces being used (I am using the soldering pads for uploading FPGA bitstream) that's somehow messing with the signals?

**Reply by ataffanel** [reply, marked as answer]
This looks like a bug. This is usually filtered by the crazyflie and we have seen that at some angle we do not see any measurements. You likely isolated the signal that are being blocked by our filter.

The first four hits looks correct, the next four are not. So my advice would be to try to filter the bad hit, for example requiring more than one identified pulse in a sweep would work here. 

Please tell us how it goes and if a simple filtering works. I am not sure why the FPGA is behaving that way but my assumption is that something goes wrong with the logic when two pulses are too close to each other.

**Reply by apark2459** [reply]
Thanks for the suggestion! I will try that and let you know how it goes.

**Reply by apark2459** [reply]
I tried setting the offset equal to zero if the frame has no channel (since the offsets with no channel seem to be way off) and this seems to work. I am getting a much higher measurement rate!
